### PR TITLE
2.12.1 and kv key validation fix

### PR DESCRIPTION
**Pull request by scottf**
1. bumping the version
2. proper kv key name check
3. handling that Nats-Expected-Last-Subject-Sequence value of zero has meaning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ The example does not work correctly against server versions after server v2.3.4
 due to a significant change made to correct _queue_ behavior that was considered wrong.
 
 If you want to take advantage of the fixes and features provided in the server after v2.3.4, 
-you __must__ upgrade to the upcoming client release, version 2.12.0
+you __must__ upgrade to the release version 2.12.0 or later.
```
Comment (scottf) [inline README.md:36]: better doc

File: README.md [hunk 2]
```diff
@@ -77,9 +77,9 @@ The java-nats client is provided in a single jar file, with a single external de
 
 ### Downloading the Jar
 
-You can download the latest jar at [https://search.maven.org/remotecontent?filepath=io/nats/jnats/2.12.0/jnats-2.12.0.jar](https://search.maven.org/remotecontent?filepath=io/nats/jnats/2.12.0/jnats-2.12.0.jar).
+You can download the latest jar at [https://search.maven.org/remotecontent?filepath=io/nats/jnats/2.12.1/jnats-2.12.1.jar](https://search.maven.org/remotecontent?filepath=io/nats/jnats/2.12.1/jnats-2.12.1.jar).
```
Comment (scottf) [inline README.md:80]: version updating from 2.12.0 to 2.12.1 - this is a common change in this PR

File: README.md [hunk 3]
```diff
@@ -136,7 +136,7 @@ If you need the absolute latest, before it propagates to maven central, you can
 ```xml
 <repositories>
     <repository>
-        <id>latest-repo</id>
+        <id>sonatype releases</id>
```
Comment (scottf) [inline README.md:139]: clearer documentation

File: src/main/java/io/nats/client/PublishOptions.java [hunk 4]
```diff
@@ -215,7 +215,8 @@ public Builder expectedLastMsgId(String lastMsgId) {
          * @return builder
          */
         public Builder expectedLastSequence(long sequence) {
-            expectedLastSeq = validateGtZeroOrMinus1(sequence, "Last Sequence");
+            // 0 has NO meaning to expectedLastSequence but we except 0 b/c the sequence is really a ulong
```
Comment (scottf) [inline src/main/java/io/nats/client/PublishOptions.java:218]: server handles 0 as default

File: src/main/java/io/nats/client/PublishOptions.java [hunk 5]
```diff
@@ -225,7 +226,7 @@ public Builder expectedLastSequence(long sequence) {
          * @return builder
          */
         public Builder expectedLastSubjectSequence(long sequence) {
-            expectedLastSubSeq = validateGtZeroOrMinus1(sequence, "Last Subject Sequence");
+            expectedLastSubSeq = validateGtEqMinus1(sequence, "Last Subject Sequence");
```
Comment (scottf) [inline src/main/java/io/nats/client/PublishOptions.java:229]: 0 has meaning for this value

File: src/main/java/io/nats/client/support/Validator.java [hunk 6]
```diff
@@ -47,11 +47,11 @@ public static String validateJetStreamPrefix(String s) {
     }
 
     public static String validateBucketNameRequired(String s) {
-        return validateLettersDigitsDashUnderscore(s, "Bucket name", true);
+        return validateKvBucketName(s, "Bucket name", true);
```
Comment (scottf) [inline src/main/java/io/nats/client/support/Validator.java:50]: better name

File: src/main/java/io/nats/client/support/Validator.java [hunk 7]
```diff
@@ -118,16 +118,23 @@ public static String validatePrintableExceptWildDotGt(String s, String label, bo
         });
     }
 
-    public static String validateLettersDigitsDashUnderscore(String s, String label, boolean required) {
+    public static String validateKvBucketName(String s, String label, boolean required) {
```
Comment (scottf) [inline src/main/java/io/nats/client/support/Validator.java:121]: better name

File: src/main/java/io/nats/client/support/Validator.java [hunk 8]
```diff
@@ -118,16 +118,23 @@ public static String validatePrintableExceptWildDotGt(String s, String label, bo
...
         return _validate(s, required, label, () -> {
-            if (notLettersDigitsDashOrUnderscore(s)) {
-                throw new IllegalArgumentException(label + " must be in the printable ASCII range and cannot include `*`, `.` or `>` [" + s + "]");
+            if (notRestrictedTerm(s)) {
```
Comment (scottf) [inline src/main/java/io/nats/client/support/Validator.java:123]: better name

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.

**Pull request merged**